File: PetStoreClient/petStore/src/app/models/product.model.ts

export class Product{

    productId: number | null;
    name: string | null;
    code: string | null;
    price: number | null;
    quantityAvailable: number | null;
    description: string | null;

    productCategoryId: number | null;
    productCategoryName: string | null;

    constructor(productId?: number, name?: string, code?: string, price?: number, quantityAvailable?: number, description?: string, productCategoryId?: number, productCategoryName?: string){
        this.productId = productId || null;
        this.name = name || null;
        this.code = code || null;
        this.price = price || null;
        this.quantityAvailable = quantityAvailable || null;
        this.description = description || null;
        this.productCategoryId = productCategoryId || null;
        this.productCategoryName = productCategoryName || null;
    }

}

File: PetStore/Models/ProductModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace PetStore.Models
{
    public class ProductModel
    {
        public int ProductId { get; set; }
        public string Name { get; set; }
        public string Code { get; set; }

        public decimal Price { get; set; }

        public int QuantityAvailable { get; set; }
        public string Description { get; set; }

        public int ProductCategoryId { get; set; }
        public string ProductCategoryName { get; set; }
    }
}
